Use 'cobalt' instead of 'content_shell' for logging and paths.

jellefoks opened the pull request:
This changes the paths and the default log file to 'cobalt' instead of
'content_shell'.

With this PR, the local storage for the `cobalt` target on linux will be
`~/.config/cobalt`

b/375193856
b/374191454

jellefoks commented:
Followup to https://github.com/youtube/cobalt/pull/4277 will make Ready for review when that merges.

andrewsavage1 commented on build/build_config.h:
```diff
@@ -254,13 +254,13 @@
 #endif
 
-#if defined(IS_COBALT)
+#if defined(ENABLE_BUILDFLAG_IS_COBALT)
```
I submitted this change, could you rebase your PR so this diff disappears?

jellefoks replied:
Done

andrewsavage1 commented on base/threading/thread_restrictions.h:
```diff
@@ -211,6 +211,11 @@ namespace chrome_cleaner {
 }  // namespace chrome_cleaner
+#if BUILDFLAG(IS_COBALT)
+namespace cobalt {
+class CobaltPathProvider;
```
@niranjanyardi Did you look into a starboard-centric way this should be done? Should we create a follow up bug for you to look at potentially changing this or are you fine with this solution? (either way, I think it should be submitted as is)

jellefoks replied:
The needed changes here is what I brought up in the chat, and this was the suggested approach. I'm not aware of a followup. This mirrors occurences of `ShellPathProvider` in this file, plus the `#if` to mark our changes to help with rebasing.

yell0wd0g replied:
My understanding of the dependencies and of the chat is that it'd make more sense to me to see USE_STARBOARD here, and define the paths in Starboard.

jellefoks replied:
As I said in the chat: This is not something needed for Starboard, but for Cobalt. You can currently build cobalt without Starboard, and changing that here would break that unnecessarily 

Now sometime in the future that, the paths may likely be provided by Starboard, and then the class can be renamed to StarboardPathProvider and this can be changed. However, that is currently not the case.

andrewsavage1 commented on cobalt/cobalt_paths.h:
```diff
@@ -0,0 +1,29 @@
+#ifndef CONTENT_SHELL_BROWSER_SHELL_PATHS_H_
+#define CONTENT_SHELL_BROWSER_SHELL_PATHS_H_
+
+#include "build/build_config.h"
```
Why is this include needed here?

jellefoks replied:
This was copied from content/shell/browser/shelL_paths.h, but it doesn't need it. removed.

andrewsavage1 commented on cobalt/cobalt_paths.cc:
```diff
@@ -0,0 +1,71 @@
+#include "base/threading/thread_restrictions.h"
+#include "build/build_config.h"
+
+#if BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
```
I don't think we need to worry about CHROMEOS buildflags

jellefoks replied:
Stripped.

andrewsavage1 commented on cobalt/cobalt_paths.h:
```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 The Chromium Authors
```
I'm not 100% on this, but I assume we'd want a Cobalt copyright on the forked code

jellefoks replied:
These files are modified shell_paths.h and shell_paths.cc with that license. Even though they are kind of trivial files, the license text should probably remain the same. What exact change to the header are you proposing here?

andrewsavage1 commented on cobalt/cobalt_paths.h:
```diff
@@ -0,0 +1,27 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CONTENT_SHELL_BROWSER_SHELL_PATHS_H_
```
I assume we want to rename the header guard here

jellefoks replied:
Done

andrewsavage1 commented on cobalt/cobalt_main_delegate.cc:
```diff
@@ -13,7 +13,41 @@
+#include "content/public/common/content_switches.h"
+#include "content/shell/common/shell_switches.h"
+
+#if !BUILDFLAG(IS_ANDROID)
```
How are these headers used? I'm confused because I don't see any linux-specifc code added, just android-specific code

jellefoks replied:
`content_switches.h` is needed for `switches::kProcessType`, and `shell_switches.h` for `switches::IsRunWebTestsSwitchPresent()`, and those are referred to here mirroring the logic in `shell_main_delegate.cc` because I don't want to incidentally break the corresponding functionality.

andrewsavage1 commented on cobalt/cobalt_paths.h:
```diff
@@ -0,0 +1,27 @@
+  // Directory where user data can be written.
+  SHELL_DIR_USER_DATA = SHELL_PATH_START,
+
+  // TODO(jam): move from content/common since it's test only.
```
Should we get rid of the TODO comment from upstream?

jellefoks replied:
Done

andrewsavage1 reviewed (commented):
This LGTM, but it would be nice to have someone like John or Miguel also take a look as I'm not familiar with some of the initialization code we're overriding

jellefoks committed:
Use 'cobalt' instead of 'content_shell' for logging and paths.

This changes the paths and the default log file to 'cobalt' instead of
'content_shell'.

With this PR, the local storage for the cobalt target on linux will be
~/.config/cobalt

b/375193856
b/374191454

    #	build_android-arm_qa.out
